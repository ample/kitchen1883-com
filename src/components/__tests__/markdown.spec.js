import React from "react"
import renderer from "react-test-renderer"

import Markdown from "../markdown"

const validNode = {
  childMarkdownRemark: {
    html: "<p>Hello World</p>",
  },
}

describe("Markdown", () => {
  it("renders null for empty objects", () => {
    const tree = renderer.create(<Markdown node={{}} />).toJSON()
    expect(tree).toBeNull()
    expect(tree).toMatchSnapshot()
  })
  it("renders null for incorrectly structured objects", () => {
    const tree = renderer.create(<Markdown node={{ hello: "world" }} />).toJSON()
    expect(tree).toBeNull()
    expect(tree).toMatchSnapshot()
  })
  it("renders correctly", () => {
    const tree = renderer.create(<Markdown node={validNode} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
