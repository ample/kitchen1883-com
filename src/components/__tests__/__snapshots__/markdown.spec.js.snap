// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown renders correctly 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>Hello World</p>",
    }
  }
/>
`;

exports[`Markdown renders null for empty objects 1`] = `null`;

exports[`Markdown renders null for incorrectly structured objects 1`] = `null`;
